src/uu/sort/src/sort.rs

```diff
@@ -1037,6 +1084,116 @@ fn get_rlimit() -> UResult<usize> {
 
 const STDIN_FILE: &str = "-";
 
+fn default_merge_batch_size() -> usize {
+    #[cfg(target_os = "linux")]
+    {
+        match get_rlimit() {
+            Ok(limit) => limit.saturating_div(4).clamp(32, 256),
+            Err(_) => 64,
+        }
+    }
+
+    #[cfg(not(target_os = "linux"))]
+    {
+        64
+    }
+}
+
+fn automatic_buffer_size(files: &[OsString]) -> usize {
+    let file_hint = file_size_hint(files);
+    let mem_hint = available_memory_hint();
+
+    // Prefer the tighter bound when both hints exist, otherwise fall back to whichever hint is available.
+    match (file_hint, mem_hint) {
+        (Some(file), Some(mem)) => file.min(mem),
+        (Some(file), None) => file,
+        (None, Some(mem)) => mem,
+        (None, None) => FALLBACK_AUTOMATIC_BUF_SIZE,
+    }
+}
+
+fn file_size_hint(files: &[OsString]) -> Option<usize> {
```

Comment:
please document this 
and maybe it would make sense to move these memory functions into a specific file